src/storage/db_interface_common.py

```diff
@@ -310,3 +311,50 @@ def release_unpacking_lock(self, uid):
 
     def drop_unpacking_locks(self):
         self.main.drop_collection('locks')
+
+    def _collect_analysis_tags_from_children(self, uid):
+        children = self._fetch_children_with_tags(uid)
+        unique_tags = {}
+        for child in children:
+            self._collect_analysis_tags(child, unique_tags)
+        return unique_tags
+
+    def _collect_analysis_tags(self, file_object, analysis_tags):
+        for name, analysis in [(n, a) for n, a in file_object.processed_analysis.items() if 'tags' in a]:
```

Comment:
You had to squeeze that into one line, didn't you? :sweat_smile: 
Converting it to a generator could be a bit more efficient (omits the intermediate list creation)

```suggestion
        for name, analysis in ((n, a) for n, a in file_object.processed_analysis.items() if 'tags' in a):
```

Reply:
Sure I did. Otherwise our guy McCabe would have come knocking :wink: